### ISSUE TITLE
Help: Kafka Producer only seems to send the message after call 'get'

### ISSUE THREAD

**Issue by WesleyBatista** [dpkp/kafka-python#1394]
I think that there is an issue regarding how the `KafkaProducer.send` works.

Trying to produce some messages from localhost to my kafka server didn't worked as expected.
I was using the very simple code example:
```python
import os
from kafka import KafkaProducer
from settings import BOOTSTRAP_SERVERS

producer = KafkaProducer(bootstrap_servers=BOOTSTRAP_SERVERS)

print("connected to: {}".format(BOOTSTRAP_SERVERS))

for _ in range(10):
    print(_)
    producer.send('foobar', b'some_message_bytes %d' % _)
```

This was only printing the results:
```
connected to: <host>:<port>
0
1
2
3
4
5
6
7
8
9
```
In addition, in another terminal panel, I was watching to the topic inside the kafka server with the following command: `/opt/bitnami/kafka/bin/kafka-console-consumer.sh --bootstrap-server=<host>:9092 --topic foobar`. No data.

Spent a lot of time tweaking configurations in the kafka server, guessing that was an issue with the Kafka `server.properties` or zookeeper configs, when finally though about to try with node js.

Using kafka-node I was able to send the messages.
```js
const kafka = require("kafka-node");

var client = new kafka.KafkaClient({kafkaHost: '<host>:<port>'}),
    producer = new kafka.Producer(client);

producer.send([{
    topic: "foobar", messages: "bla bla bla"
}], function(err, data){console.log(err, data)})
```

Then I knew that should be something with the library. 

So I tried to call 'get' after the 'send':
```python
    g = producer.send('foobar', b'some_message_bytes %d' % _)
    print(g.get())
```

After call "get" after each "send" I could finally see the messages getting sent.
```
RecordMetadata(topic='foobar', partition=0, topic_partition=TopicPartition(topic='foobar', partition=0), offset=32, timestamp=1519618473862, checksum=None, serialized_key_size=-1, serialized_value_size=21)
48
RecordMetadata(topic='foobar', partition=0, topic_partition=TopicPartition(topic='foobar', partition=0), offset=33, timestamp=1519618474007, checksum=None, serialized_key_size=-1, serialized_value_size=21)
49
RecordMetadata(topic='foobar', partition=0, topic_partition=TopicPartition(topic='foobar', partition=0), offset=34, timestamp=1519618474148, checksum=None, serialized_key_size=-1, serialized_value_size=21)
```

I think that we shouldn't need to call another method after 'send' to have the messages being delivered..

Have I done something wrong?

**Comment by dpkp** [comment]
Try calling `producer.flush()` and/or `producer.close()` before you exit your test script. Also try simply running your commands from a repl. It seems to me that you are simply shutting down your python interpreter before the KafkaProducer background thread has a chance to batch up your messages and send them over the network.

**Comment by WesleyBatista** [comment]
You are right. I was testing the node example in the interactive mode, while with the python example I was testing calling `python path/to/script.py` in the command line.
When I tried put the node code in a file and call `node path/to/script.js`, the messages was sent, but the node hanged (didn't closed the interpreter). So I tried to put a `process.exit()` at the end of my nodejs script, and as expected, the message wasn't sent.

Here is a clear example about the importance of testing things in the same way.

Thanks for the tip :)
